### DISCUSSION TITLE
Add a vector art mode to find_trim

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/libvips#2076] [Ideas]
`find_trim` is designed for photographic images and the median filter can remove single-pixel elements in vector art sources.

Add a vector art mode, and improve the speed too by going to monochrome early in processing.

Referring issue: https://github.com/lovell/sharp/issues/2363

**Comment by miltoncandelero** [comment]
any news on this?
Do we have any way of trimming the alpha around an image without losing pixels?
Ref: https://github.com/lovell/sharp/issues/2363

**Reply by miltoncandelero** [reply]
@jcupitt I found the offending line and it seems kinda dumb

It is doing a median of the image... why?

https://github.com/libvips/libvips/blob/bae0342d9506a4e6f2e5e8ba0bb612c6a601044e/libvips/arithmetic/find_trim.c#L124

(and it's documented to do so)
https://github.com/libvips/libvips/blob/bae0342d9506a4e6f2e5e8ba0bb612c6a601044e/libvips/arithmetic/find_trim.c#L255

**Reply by jcupitt** [reply]
Hi @miltoncandelero,

From memory, you need the median for (for example) JPEG images, since it'll trigger on compression artefacts otherwise.

There was a blog post about this around a million years ago:

http://libvips.blogspot.com/2014/02/trim-auto-crop-with-ruby-vips.html

**Reply by miltoncandelero** [reply]
but then we have a feature that should be called `find_trim_in_artifacty_jpeg` instead of `find_trim` 🙃 

Right now it seems is doing something more akin to "point of interest" than "trim" 🤔

**Reply by jcupitt** [reply]
Any imaging source will have noise and need a median filter, it's not just JPEGs. A PNG image scanned from a photo will need a median too.

I think we just need to add a flag to tune it for trimming synthetic images, perhaps `synthetic` or maybe `no-filter`? It's only a couple of lines of code (and some testing).

**Reply by jcupitt** [reply]
... I made a tiny PR for this feature https://github.com/libvips/libvips/pull/3329

testing very welcome etc.
